junit-jupiter-api/src/main/kotlin/org/junit/jupiter/api/Assertions.kt

```diff
@@ -86,6 +101,139 @@ fun assertAll(vararg executables: () -> Unit) =
 fun assertAll(heading: String?, vararg executables: () -> Unit) =
     assertAll(heading, executables.toList().stream())
 
+/**
+ * Example usage:
+ * ```kotlin
+ * val string: String? = ...
+ *
+ * assertNull(string)
+ *
+ * // compiler won't allow even safe calls, since the string is always null
+ * // string?.isNotEmpty()
+ * ```
+ * @see Assertions.assertNull
+ */
+@API(since = "5.10", status = EXPERIMENTAL)
+fun assertNull(actual: Any?) {
+    contract {
+        returns() implies (actual == null)
+    }
+
+    Assertions.assertNull(actual)
+}
+
+/**
+ * Example usage:
+ * ```kotlin
+ * val string: String? = ...
+ *
+ * assertNull(string, "Should be nullable")
+ *
+ * // compiler won't allow even safe calls, since the string is always null
+ * // string?.isNotEmpty()
+ * ```
+ * @see Assertions.assertNull
+ */
+@API(since = "5.10", status = EXPERIMENTAL)
+fun assertNull(actual: Any?, message: String) {
+    contract {
+        returns() implies (actual == null)
+    }
+
+    Assertions.assertNull(actual, message)
+}
+
+/**
+ * Example usage:
+ * ```kotlin
+ * val string: String? = ...
+ *
+ * assertNull(string) { "Should be nullable" }
+ *
+ * // compiler won't allow even safe calls, since the string is always null
+ * // string?.isNotEmpty()
+ * ```
+ * @see Assertions.assertNull
+ */
+@API(since = "5.10", status = EXPERIMENTAL)
+fun assertNull(actual: Any?, messageSupplier: () -> String) {
+    contract {
+        returns() implies (actual == null)
+
+        callsInPlace(messageSupplier, AT_MOST_ONCE)
+    }
+
+    Assertions.assertNull(actual, messageSupplier)
+}
+
+/**
+ * Example usage:
+ * ```kotlin
+ * val string: String? = ...
+ *
+ * assertNotNull(string)
+ *
+ * // compiler smart casts nullableString to a non-nullable object
+ * assertTrue(string.isNotEmpty())
+ * ```
+ * @see Assertions.assertNotNull
+ */
+@API(since = "5.10", status = EXPERIMENTAL)
+fun <T> assertNotNull(actual: T?): T {
```

Comment:
This method has a return value, that'll be unused most of the time because of smart casts. For consistency with Java API, maybe don't return here?

Reply:
Same for other overloads

Reply:
Agree. There is no point in returning a value here

Reply:
This is resolved.